tools/site-query/scripts.js

```diff
@@ -312,22 +306,28 @@ async function init(doc) {
       const caption = table.querySelector('caption');
       caption.setAttribute('aria-hidden', false);
       caption.querySelector('.term').textContent = query;
-      caption.querySelector('.results-found').textContent = 0;
-      caption.querySelector('.results-of').textContent = 0;
+      const resultsFoundElement = caption.querySelector('.results-found');
+      resultsFoundElement.textContent = 0;
+      const resultsOfElement = caption.querySelector('.results-of');
+      resultsOfElement.textContent = 0;
 
+      const processingTasks = [];
       // eslint-disable-next-line no-restricted-syntax
       for await (const sitemapUrl of sitemapUrls) {
         if (sitemapUrl.pathname.startsWith(path)) {
           searched += 1;
-          processUrl(sitemapUrl, query, queryType, org, site).then((tr) => {
-            if (tr) {
-              results.append(tr);
-            }
-          });
+          const promise = processUrl(sitemapUrl, query, queryType, org, site)
+            .then((tr) => {
+              if (tr) {
+                results.append(tr);
+                resultsFoundElement.textContent = results.children.length;
+              }
+            });
+          processingTasks.push(promise);
         }
-        updateCaption(caption, results.children.length, searched);
       }
-      updateCaption(caption, results.children.length, searched);
+      resultsOfElement.textContent = searched;
```

Comment:
by doing this here, the number searched now only updates once at the end. Can you move this line inside of the `then` function, to be more inline with how it worked previously?

Reply:
Done.